### ISSUE TITLE
GT angle must be in range [-pi/2, pi/2]？

### ISSUE THREAD

**Issue by Cappuccinoll** [open-mmlab/mmrotate#808]
### Branch

1.x branch https://mmrotate.readthedocs.io/en/1.x/

### 📚 The doc issue

hi

when i use the phase shifting coder, i notice that oient is in the interbal [-pi/2, pi/2]，that means the input angle in dataset must be limited in range [-pi/2, pi/2]?

thanks!

Best regards

**Comment by zytx121** [comment]
@yuyi1005, Could you please answer this question?

**Comment by yuyi1005** [comment]
The default code is limited to [-pi/2, pi/2]. But in dual-freq PSC, it would be [-pi/2, pi/2] for one frequency and [-pi/4, pi/4] for the other (for square like objects, -pi/4 is equivalent to pi/4 , so [-pi/4, pi/4] is enough). You can simply add an offset to the angle if you want something like [-pi/4, 3pi/4].

**Comment by Februry** [comment]
> 

if the input and output angle are both in [-pi, pi], should the encoder be like:
```
phase_targets = angle_targets * 1
if self.dual_freq:
    phase_targets = angle_targets * 2
```

**Comment by yuyi1005** [comment]
Yep, the first phase is supposed to be phase_targets = angle_targets * 1 for [-pi, pi] heading detection. If all the objects have a "head", then single frequency is enough. But when some objects are 180-rotated symmetry (eg. Tennis Court), dual frequency can be used as phase_targets = angle_targets * 2, which outputs [-pi / 2, pi / 2]

**Comment by Februry** [comment]
> 

@yuyi1005 thanks for the reply, for vehicle detection task, all the objects do have a "head", which refers to their heading directions, but some special vehicles can be square boxes, should i use single frequency or dual frequency? what should the encoder and decoder be like in my case?

**Comment by yuyi1005** [comment]
I would recommend tring the single frequency one first
encode: phase_targets = angle_targets * 1 
decode: angle_pred = phase / 1
